helix-core/src/main/java/org/apache/helix/cloud/AzureInformationFetcher.java

```diff
@@ -0,0 +1,36 @@
+package org.apache.helix.cloud;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import org.apache.helix.common.cloud.CloudInformationFetcher;
+
+public class AzureInformationFetcher implements CloudInformationFetcher {
```

Comment:
VMInformationFetcher? or more generic, InstanceInformationFetcher?

Reply:
Please see above comment. I'm thinking whether we should keep it more general than VM/Instance, so later if we want to get other field's information, we can also use this function.

Reply:
Rethink about it, and this API is actually called instance metadata service. Maybe we should have something similar in the name.